p2p/connection.go

```diff
@@ -553,29 +553,27 @@ func (chDesc *ChannelDescriptor) FillDefaults() {
 type Channel struct {
 	conn          *MConnection
 	desc          *ChannelDescriptor
-	id            byte
 	sendQueue     chan []byte
 	sendQueueSize int32 // atomic.
 	recving       []byte
 	sending       []byte
-	priority      int
 	recentlySent  int64 // exponential moving average
 
 	maxMsgPacketPayloadSize int
 }
 
 func newChannel(conn *MConnection, desc *ChannelDescriptor) *Channel {
+	descCopy := *desc
+	desc = &descCopy
 	desc.FillDefaults()
 	if desc.Priority <= 0 {
 		cmn.PanicSanity("Channel default priority must be a postive integer")
 	}
 	return &Channel{
 		conn:                    conn,
 		desc:                    desc,
-		id:                      desc.ID,
```

Comment:
shouldn't we remove `id` and `priority` from the struct itself too ?

Reply:
@ebuchman i thought i did that already here https://github.com/tendermint/tendermint/pull/831/files#diff-b142572faedb96d82e71edd098a690b0L556 and a few lines below but maybe i forgot something or did you misread maybe?

Reply:
yep not sure how I missed that !